sled-agent/src/services.rs

```diff
@@ -495,65 +806,67 @@ impl ServiceManager {
     // more clearly.
     pub async fn load_services(&self) -> Result<(), Error> {
         let log = &self.inner.log;
-        let ledger_paths = self.all_service_ledgers().await;
-        info!(log, "Loading services from: {ledger_paths:?}");
-
         let mut existing_zones = self.inner.zones.lock().await;
         let Some(mut ledger) =
-            Ledger::<AllZoneRequests>::new(log, ledger_paths).await
+            self.load_ledgered_zones(&existing_zones).await?
         else {
-            info!(log, "Loading services - No services detected");
+            // Nothing found -- nothing to do.
+            info!(
+                log,
+                "Loading Omicron zones - \
+                no zones nor old-format services found"
+            );
             return Ok(());
         };
-        let services = ledger.data_mut();
+
+        let zones_config = ledger.data_mut();
+        info!(
+            log,
+            "Loaded Omicron zones";
+            "zones_config" => ?zones_config
+        );
+        let omicron_zones_config =
+            zones_config.clone().to_omicron_zones_config();
```

Comment:
We definitely could...but then if we had more consumers that didn't need to clone, wouldn't we want to switch this back?  I'm not clear on what principles guide a lot of the choices around these types of functions (but I think it also doesn't matter too much unless the runtime cost is significant).

Reply:
This is way down in the weeds, and not material to most of the PR, but just thinking out loud (heavily based on https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv): I think there are three really obvious cases:

1. `as_foo(&self) -> &Foo` should be completely free (e.g., just borrowing some field within `self`)
2. `into_foo(self) -> Foo` when it's just moving values (e.g., taking ownership of some field within `self`)
3. `to_foo(&self) -> Foo` when it's _not_ free; we're going to clone or allocate or something to produce a `Foo`

The guidelines note that the latter two aren't as strict as I think of them:

* `info_foo(self) -> Foo` is allowed to do more expensive things than just moving values (their example is `BufWriter::into_inner()`, which flushes data prior to returning)
* `to_foo` usually takes `&self`, but is allowed to take `self` if `Self: Copy`. It's assumed to not be free in either case; some new work or allocation has to happen to produce the new `Foo`.

I think in this case there are two reasonable choices:

1. `into_omicron_zones_config(self)`: basically what we have now, but renamed to `into`. This is okay by the guidelines, but I don't prefer it: it isn't the "just moving values" case because internally it collects and allocates a new `Vec`.
2. `to_omicron_zones_config(&self)`: would be my vote

The current `to_omicron_zones_config(self)` is technically not following the guidelines, I think, because `to_*` for owned -> owned should imply that the types are `Copy`, which they are not.